app/helpers/navigation_sidebar_helper.rb

```diff
@@ -0,0 +1,11 @@
+module NavigationSidebarHelper
+  def construct_sidebar_link(link, section)
```

Comment:
It would be good to follow the naming pattern of the underlying Rails tag helper
Maybe something like `sidebar_link_to` ?
Alternatively as the link `:class` and `:rel` attrs are the only thing that varies you could use `link_to` and a helper method for the attributes.